src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/tests/test_query_injection.py

```diff
@@ -63,11 +99,12 @@ def test_string_replacement(self):
         }
         flag = self.shell.handle_jmespath_query(args)
         self.assertTrue(flag)
-        self.assertEqual(u"vm show -g mygroup -n myname\n", self.stream.getvalue())
+        results = self.stream.getvalue().split('\n')
```

Comment:
using `os.sep` (https://docs.python.org/2/library/os.html#os.sep) instead of `\n`. Remember the newline separator is different on the different platforms (Windows vs Linux/Unix).

Reply:
os.linesep? the `\n` character was being written to the stream directly in _mock_execute() so this might have been okay. Anyhow, I subbed in `os.linesep` for `\n` and decoded the json strings back(I printed them using json.dumps) to use in asserts.